src/deploymentSlotActions.ts

```diff
@@ -76,21 +81,20 @@ class SwapStep extends WizardStep {
             this.targetSlot = result.label;
             const credential = this.azureAccount.getCredentialByTenantId(this.slot.subscription.tenantId);
             const client = new WebSiteManagementClient(credential, this.slot.subscription.subscriptionId);
-
-            await client.webApps.swapSlotSlot(this.slot.site.resourceGroup, this.slot.site.repositorySiteName, { targetSlot: this.targetSlot, preserveVnet: true }, this.sourceSlot.label, {},
-                (err, _result, _request, response) => {
-                    if (!err && response.statusCode === 200) {
-                        this.wizard.writeline(`"${this.targetSlot}" was swapped with "${this.sourceSlot.label}"`);
-                    } else {
-                        // the format of Azure error messages are JSON strings
-                        this.wizard.writeline(JSON.parse(err.message).Message);
-                    }
-                });
+            try {
+                this.targetSlot === 'production' ?
+                    await client.webApps.swapSlotWithProductionWithHttpOperationResponse(this.slot.site.resourceGroup, this.slot.site.repositorySiteName, { targetSlot: this.sourceSlot.label, preserveVnet: true }) :

```

Comment:
Is there a `swapSlotWithProduction` method?

Not sure why the 'WithHttpOperationResponse' is used